src/pytorch_lightning/plugins/io/async_plugin.py

```diff
@@ -0,0 +1,75 @@
+# Copyright The PyTorch Lightning team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import queue
+from typing import Any, Dict, Optional
+
+from pytorch_lightning.plugins.io.checkpoint_plugin import CheckpointIO
+from pytorch_lightning.plugins.io.wrapper import _WrappingCheckpointIO
+from pytorch_lightning.utilities.cloud_io import _ThreadQueue
+from pytorch_lightning.utilities.types import _PATH
+
+
+class AsyncCheckpointIO(_WrappingCheckpointIO, CheckpointIO):
+    """AsyncCheckpointIO enablses saving the checkpoints asynchronously.
+
+    .. warning::
+
+        This is currently an experimental plugin/feature and API changes are to be expected.
+
+    Args:
+        checkpoint_io: A checkpoint IO plugin that is used as the basis for async checkpointing.
+        interval: Sleep time between each queue check.
+    """
+
+    def __init__(self, checkpoint_io: Optional["CheckpointIO"] = None, interval: float = 2.0) -> None:
+        super().__init__(checkpoint_io)
+
+        self._thread = _ThreadQueue(q=queue.Queue(), interval=interval)
+        self._thread.start()
```

Comment:
I recommend not starting the thread on init, but dynamically on the first call to save_checkpoint().
This allows us to use this with multiprocessing and don't need to worry about locks or threads not being picklable or forkable.

Reply:
done!